docs/reference/aggregations/bucket/variablewidthhistogram-aggregation.asciidoc

```diff
@@ -0,0 +1,85 @@
+[[search-aggregations-bucket-variablewidthhistogram-aggregation]]
+=== Variable Width Histogram Aggregation
+
+This is a multi-bucket aggregation similar to <<search-aggregations-bucket-histogram-aggregation>>.
+However, the width of each bucket is not specified. Rather, a target number of buckets is provided and bucket intervals
+are dynamically determined based on the document distribution. This is done using a simple one-pass document clustering algorithm
+that aims to obtain low distances between bucket centroids. Unlike other multi-bucket aggregations, the intervals will not
+necessarily have a uniform width.
+
+TIP: The number of buckets returned will always be less than or equal to the target number.
+
+Requesting a target of 3 buckets.
+
+[source,console]
+--------------------------------------------------
+POST /sales/_search?size=0
+{
+    "aggs" : {
+        "prices" : {
+            "variable_width_histogram" : {
+                "field" : "price",
+                "buckets" : 2
+            }
+        }
+    }
+}
+--------------------------------------------------
+// TEST[setup:sales]
+
+Response:
+
+[source,console-result]
+--------------------------------------------------
+{
+    ...
+    "aggregations": {
+        "prices" : {
+            "buckets": [
+                {
+                    "key": 10.0,
+                    "doc_count": 2
+                },
+                {
+                    "key": 150.0,
+                    "doc_count": 5
+                }
+            ]
+        }
+    }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/\.\.\./"took": $body.took,"timed_out": false,"_shards": $body._shards,"hits": $body.hits,/]
+
+==== Shard size
+The `shard_size` parameter specifies the number of buckets that the coordinating node will request from each shard.
+A higher `shard_size` leads each shard to produce smaller buckets. This reduce the likelihood of buckets overlapping
+after the reduction step. Increasing the `shard_size` will improve the accuracy of the histogram, but it will
+also make it more expensive to compute the final result because bigger priority queues will have to be managed on a
+shard level, and the data transfers between the nodes and the client will be larger.
+
+TIP: Parameters `buckets`, `shard_size`, and `cache_limit` are optional. By default, `buckets = 10`, `shard_size = 500` and `cache_limit = min(50 * shard_size, 50000)`.
+
+==== Cache Limit
+The `cache_limit` parameter can be used to specify the number of individual documents that will be stored in memory
```

Comment:
I wonder if `initial_buffer` or something would be a better name. This doesn't fit the definition of "cache" that I'm used to.